pandas/core/groupby/generic.py

```diff
@@ -1247,14 +1275,16 @@ def _wrap_applied_output(self, keys, values, not_indexed_same=False):
             # make Nones an empty object
             if first_not_none is None:
                 return self.obj._constructor()
-            elif isinstance(first_not_none, NDFrame):
+            elif isinstance(first_not_none, (Series, DataFrame)):
 
                 # this is to silence a DeprecationWarning
                 # TODO: Remove when default dtype of empty Series is object
                 kwargs = first_not_none._construct_axes_dict()
+                backup: FrameOrSeriesUnion
                 if isinstance(first_not_none, Series):
+                    kwargs["dtype_if_empty"] = object
```

Comment:
Also for mypy. I think this is better though (in case `_construct_axes_dict` had a `dtype_if_empty` key).